pallets/gear/src/manager/mod.rs

```diff
@@ -377,7 +383,15 @@ where
         ProgramStorageOf::<T>::remove_program_pages(program_id, memory_infix);
 
         let program_account = program_id.cast();
-        let balance = CurrencyOf::<T>::free_balance(&program_account);
+
+        // Only `reducible_balance` is allowed to be transferred out, even if a part of the
```

Comment:
does reducible balance include ED?

Reply:
Depends on `Preservation` requirement. If `Preservation::Expendable` is supplied then the account is allowed to die therefore the ED is included in the `reducible_balance`. Otherwise it's not